stan/math/prim/scal/fun/log1m.hpp

```diff
@@ -33,10 +33,15 @@ namespace math {
  *
  * @param[in] x Argument.
  * @return Natural log of one minus the argument.
- * @throw std::domain_error If the argument is greater than 1.
- * @throw std::overflow_error If the computation overflows.
+ * @throw <code>std::domain_error</code> If the argument is greater than 1.
+ * @throw <code>std::overflow_error</code> If the computation overflows.
  */
-inline double log1m(double x) { return stan::math::log1p(-x); }
+inline double log1m(double x) {
+  if (x > 1)
+    domain_error("Error in function log1m(double)", "log1m(x)", x,
```

Comment:
rather than `domain_error`, please use `check_less_or_equal()`. See the source here: <https://github.com/stan-dev/math/blob/develop/stan/math/prim/scal/err/check_less_or_equal.hpp>

Also see tests for usage: <https://github.com/stan-dev/math/blob/develop/test/unit/math/prim/scal/err/check_less_or_equal_test.cpp>